src/vs/workbench/contrib/search/common/searchNotebookHelpers.ts

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+
+import { FindMatch, IReadonlyTextBuffer } from 'vs/editor/common/model';
+import { TextSearchMatch, IFileMatch, ITextSearchMatch } from 'vs/workbench/services/search/common/search';
+import { Range } from 'vs/editor/common/core/range';
+import { URI, UriComponents } from 'vs/base/common/uri';
+
+export type IRawClosedNotebookFileMatch = INotebookFileMatchNoModel<UriComponents>;
+
+export interface INotebookFileMatchNoModel<U extends UriComponents = URI> extends IFileMatch<U> {
+	cellResults: INotebookCellMatchNoModel<U>[];
+}
+
+export interface INotebookCellMatchNoModel<U extends UriComponents = URI> {
+	index: number;
+	contentResults: ITextSearchMatch<U>[];
+	webviewResults: ITextSearchMatch<U>[];
+}
+
+export function isINotebookFileMatchNoModel(object: IFileMatch): object is INotebookFileMatchNoModel {
+	return 'cellResults' in object;
+}
+
+export function reviveINotebookCellMatchNoModel(cellMatch: INotebookCellMatchNoModel<UriComponents>): INotebookCellMatchNoModel<URI> {
+	return {
+		index: cellMatch.index,
+		contentResults: cellMatch.contentResults.map(e => {
+			return {
+				...e,
+				...{ uri: URI.revive(e.uri) }
```

Comment:
You can use the helper `revive` from marshalling.ts that walks the object and does this automatically